src/dispatchers/keyDispatcher.ts

```diff
@@ -34,30 +34,34 @@ export module Dispatchers {
 
       this._event2Callback["keydown"] = (e: KeyboardEvent) => this._processKeydown(e);
 
-      this._keydownBroadcaster = new Core.Broadcaster(this);
-      this._broadcasters = [this._keydownBroadcaster];
+      this._keydownCallbackSet = new Utils.CallbackSet();
+      this._callbackSets = [this._keydownCallbackSet];
     }
 
-    protected _getWrappedCallback(callback: Function): Core.BroadcasterCallback<Dispatchers.Key> {
-      return (d: Dispatchers.Key, e: KeyboardEvent) => callback(e.keyCode, e);
+    /**
+     * Registers a callback to be called whenever a key is pressed,
+     *
+     * @param {KeyCallback} callback
+     * @return {Dispatcher.Key} The calling Dispatcher.Key.
+     */
+    public onKeyDown(callback: KeyCallback): Key {
+      this._setCallback(this._keydownCallbackSet, callback);
+      return this;
```

Comment:
Err... looking back at the old implementation, I'm now not too confident in this switch of implementation.  Can you do a quick check to see if onKeyDown works as intended?